helpers/classes.py

```diff
@@ -0,0 +1,171 @@
+import re
+from bs4 import BeautifulSoup
+try:
+    from openedx.core.lib.xblock_serializer.api import serialize_xblock_to_olx
+except ModuleNotFoundError:
+    # This can be removed in Quince.
+    from openedx.core.djangoapps.olx_rest_api.block_serializer import XBlockSerializer as serialize_xblock_to_olx
+from helpers.config import (
+    LMS_URL,
+    STUDIO_URL,
+)
+
+class Block:
+    '''
+    Represents each block in a course
+    '''
+    def __init__(self, block_obj, parent) -> None:
+        self.usage_key = str(block_obj.location)
+        self.id = block_obj.location.block_id
+        self.obj = block_obj
+        self.studio_url = ''
+        if block_obj.category == 'vertical':
+            self.studio_url = f"{STUDIO_URL}/container/{str(block_obj.location)}"
+        self.display_name = block_obj.display_name_with_default
+        self.parent = parent
+        self.data = ''
+        if block_obj.category not in ['chapter', 'sequential', 'vertical']:
+            # We want to get the olx for all the XBlocks, but sometimes there might be
+            # exceptions while serializing. In this case, we fallback to the `data` property
+            # Using the data property is not ideal since all XBlocks do not implement this,
+            # but it should cover the most important once like html and problem XBlocks.
+            try:
+                self.data = serialize_xblock_to_olx(block_obj).olx_str
```

Comment:
This adds escapes the HTML in some XBlocks. Let's assume the following scenario, where we include a link in the [HTML5 XBlock](https://github.com/open-craft/xblock-html). The links are not detected there.

```python
In [1]: from opaque_keys.edx.locator import *
In [2]: from openedx.core.djangoapps.olx_rest_api.block_serializer import XBlockSerializer as serialize_xblock_to_olx
In [3]: from xmodule.modulestore.django import modulestore

In [4]: key = BlockUsageLocator.from_string('block-v1:etc')
In [5]: block = store.get_item(key)

In [6]: block.data
Out[6]: '<p><a href="/jump_to_id/vertical_0270f6de40fc">Hello</a></p>'

In [7]: block.data
Out[7]: '<html5 xblock-family="xblock.v1" data="&lt;p&gt;&lt;a href=&quot;/jump_to_id/vertical_0270f6de40fc&quot;&gt;Hello&lt;/a&gt;&lt;/p&gt;"/>\n'
```

This is also happening for the Drag and Drop v2 XBlock. It's not happening in the Problem XBlock.

Unless we can work around it, I believe that in our scenario using `block.data` may be sufficient. However, we will need to parse Drag and Drop v2 a bit differently, because its data is stored as a dict.

Reply:
I avoided using `block.data` in the first place because of xblocks like drag&drop and ORA, for which `block.data` did not return all configured texts/links. Also, all xblock might not provide the `data` at all. OLX proved much more realiable source for fetching all links configured in the xblock.

To handle the escaped html, I unescaped it using `html.unescape()`. This seems to working correctly for both drag&drop as well as HTML5 XBlock.

Reply:
@kaustavb12, unescape doesn't resolve everything - if I add `<a href="/jump_to_id/vertical_0270f6de40fc">Hello</a>` to the `Feedback` field of DnDv2, it still produces `(lots of text) "start": "<a href=\\"/jump_to_id/vertical_0270f6de40fc\\">Hello</a>"\n  } (lots of text)`. Would it make sense to add `.replace('\\', '')`?

Reply:
@Agrendalath  Thanks for spotting this. Fixed.